src/containers/Page/RightPage.js

```diff
@@ -4,59 +4,111 @@ import classes from "./RightPage.module.css";
 import Button from "../../components/Button/Button";
 
 class RightPage extends Component {
-  state = { firstName: "", lastName: "", email: "", password: "" };
+  state = {
+    user: { firstName: "", lastName: "", email: "", password: "" },
+    submitted: false,
+    errors: {},
+  };
 
   onChangeHandler = (event) => {
-    const key = event.target.name;
-    const value = event.target.value;
-    this.setState({ [key]: value });
-    console.log(this.state.firstName);
+    const { user } = this.state;
+    user[event.target.name] = event.target.value;
+    this.setState({ user });
+    // const key = event.target.name;
+    // const value = event.target.value;
+    // this.setState({ [key]: value });
+    // console.log(this.state.firstName);
+  };
+
+  formSubmitHandler = () => {
+    const err = {};
+    const { firstName, lastName, email, password } = this.state.user;
+    if (!firstName) {
+      err.firstName = "This field is Required";
+    }
+
+    if (!lastName) {
+      err.lastName = "This field is Required";
+    }
+    if (!email) {
+      err.email = "E-mail is Required";
+    }
+    if (password.length < 8) {
+      err.password = "Password should be more than 8 characters";
+    }
+
+    this.setState({ errors: err }, () => {
+      if (Object.getOwnPropertyNames(this.state.errors).length === 0) {
```

Comment:
You should use callbacks as rarely as possible and use async/await instead. Especially on setState